sympy/integrals/integrals.py

```diff
@@ -1242,6 +1242,110 @@ def _sage_(self):
                                     hold=True)
         return f
 
+    def principal_value(self):
+        """
+        Compute the Cauchy Principal Value of the definite integral of a real function in the given interval
+        on the real axis.
+        In mathematics, the Cauchy principal value, is a method for assigning values to certain improper
+        integrals which would otherwise be undefined.
+
+        Examples
+        ========
+
+        >>> from sympy import Dummy, symbols, integrate, limit, oo
+        >>> from sympy.integrals.integrals import Integral
+        >>> from sympy.calculus.singularities import singularities
+        >>> x = symbols('x')
+
+        >>> g = x + 1
+        >>> Integral(g, (x, -oo, oo)).principal_value()
+        nan
+
+        >>> f = 1/(x**3)
+        >>> singularities(f,x)
+        {0}
+
+        >>> Integral(f, (x, -oo, oo)).principal_value()
+        0.0
+
+        >>> Integral(f, (x, -oo, 0)).principal_value()
+        -oo
+
+        >>> Integral(f, (x, 0, oo)).principal_value()
+        oo
+
+        >>> Integral(f, (x, 1, 2)).principal_value()
+        3/8
+
+        >>> Integral(f, (x, -2, -1)).principal_value() + Integral(f, (x, 1, 2)).principal_value()
+        0
+
+        >>> Integral(f, (x, 0.4, 3)).principal_value()
+        3.06944444444444
+
+        >>> Integral(f, (x, -3, -0.4)).principal_value() + Integral(f, (x, 0.4, 3)).principal_value()
+        0
+
+
+        References
+        ==========
+
+        .. [1] http://en.wikipedia.org/wiki/Cauchy_principal_value
+        .. [2] http://mathworld.wolfram.com/CauchyPrincipalValue.html
+        """
+        from sympy.calculus import singularities
+
+        if len(self.limits) != 1 or len(list(self.limits[0])) != 3:
+            raise ValueError("You need to insert a variable, lower_limit, and upper_limit correctly to calculate "
+                             "cauchy's principal value")
+
+        x, a, b = self.limits[0]
+
+        if not(a <= b and a.is_comparable and b.is_comparable):
```

Comment:
`a <= b` should be the last test after it been checked that the comparison makes sense.

Reply:
@jksuom  Okay, Got it. Implementing now..